### PR TITLE
Issue 42 - Refactor OpenShiftServiceFactory as an CDI

### PR DESCRIPTION
**Pull request by xcoulon**
Applying the same patterns as for issue #41, using
a CDI producer to instanciate the implementation,
and use a base test class for both POJO Junit test
and integration tests.

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Comment by xcoulon** [comment]
@ALRubinger can you review this PR, please ?

File: services/openshift-service-impl/src/main/java/org/kontinuity/catapult/service/openshift/impl/fabric8/openshift/client/OpenShiftServiceProducer.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
+/**
```
Comment (ALRubinger) [inline services/openshift-service-impl/src/main/java/org/kontinuity/catapult/service/openshift/impl/fabric8/openshift/client/OpenShiftServiceProducer.java]: Wrong copyright :)

File: services/openshift-service-impl/src/main/java/org/kontinuity/catapult/service/openshift/impl/fabric8/openshift/client/OpenShiftUrl.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
+/*******************************************************************************
```
Comment (ALRubinger) [inline services/openshift-service-impl/src/main/java/org/kontinuity/catapult/service/openshift/impl/fabric8/openshift/client/OpenShiftUrl.java]: Wrong copyright

Comment (xcoulon) [inline services/openshift-service-impl/src/main/java/org/kontinuity/catapult/service/openshift/impl/fabric8/openshift/client/OpenShiftUrl.java]: Damn'it ! Indeed, I need to fix that in my workspace settings :-/

**Comment by ALRubinger** [comment]
a63338fda57172ec50869c03127468d5b1f7ce18

**Pull request closed without merging**